### DISCUSSION TITLE
SubContainer items inherit ItemContainer rotation, but cannot be independently rotated

### DISCUSSION THREAD

**Discussion by ShadowDragon8685** [FakeFishGames/Barotrauma#12248] [Code questions and modding support]
I was trying to add a flashlight add-on slot to the Syringe Gun (for example), but the base itemcontainer has a 90-degree rotation. I could **not** get the flashlight to rotate back to 0 - IE, pointing forward - no matter what I did messing with the SubContainer.

Please make SubContained items able to have an independent rotation.

    <ItemContainer capacity="1" maxstacksize="4" itempos="0,25" itemrotation="-90" hideitems="false" containedstateindicatorstyle="syringe" containedstateindicatorslot="0">

      <SlotIcon slotindex="0" texture="Content/UI/StatusMonitorUI.png" sourcerect="384,448,64,64" origin="0.5,0.5" />

	  <SlotIcon slotindex="1" texture="Content/UI/StatusMonitorUI.png" sourcerect="320,448,64,64" origin="0.5,0.5" />

      <Containable items="syringe" />

	  <SubContainer capacity="1" maxstacksize="1" >

	  

        <Containable items="flashlight" hide="false" itempos="22,-1" itemrotation="0" setactive="true" />

      </SubContainer>

    </ItemContainer>

**Comment by BiffW** [comment]
move `itemrotation="-90"` from `ItemContainer` to `Containable`
